Add custom components to mdx integration guide (#4530)
withastro/astro: packages/integrations/mdx/README.md

<!-- -->
```

### Custom components

> Under the hood, MDX will convert markdown into html components. For example, 

Reviewer: ```suggestion
Under the hood, MDX will convert markdown into html components. For example, this blockquote:
```

Small nit for clarity!

```md
> A blockquote with *some* emphasis.
```

> will be converted into

Reviewer: ```suggestion
will be converted into this HTML:
```

Nit: we usually end lines with colons for snippets.

```html
<blockquote>
  <p>A blockquote with <em>some</em> emphasis.</p>
</blockquote>
```

> MDX provides a way to tap into this rendering and use your own custom components. In the above example, you could create a custom Blockquote component (in any language) that has either a `<slot />`  component or accepts a `children` prop. Then in the MDX file you import the component and export it to the `components` export. 

Reviewer: ```suggestion
But what if you want to map these blockquotes to our own markup? In the above example, you could create a custom `Blockquote` component (in any language) that has either a `<slot />`  component or accepts a `children` prop. 

[example here?]

Then, you can map the `blockquote` element to your `Blockquote` component using the `components` export:
```

Nit on explaining the "why" before diving in. Also, I think an example `Blockquote.jsx` component snippet would clarify what "slot" and "children" would be used for!

```mdx
import Blockquote from '../components/Blockquote.astro';
export const components = { blockquote: Blockquote };
```

> The advantage of this is it allows for the simplicity of writing in markdown without having to write the custom component or writing a remark/rehype plugin. A full list of components that can have custom components is on the [MDX website](https://mdxjs.com/table-of-components/).

Reviewer: ```suggestion
The advantage of this is it allows for the simplicity of writing in markdown without having to write the custom component or writing a remark/rehype plugin. Visit the [MDX website](https://mdxjs.com/table-of-components/) for a full list of HTML elements that support custom components.
```

Nit: components -> HTML elements for clarity

#### Custom components with imported `mdx`

> Custom components can also be passed to the components prop when rending imported MDX content.

Reviewer: ```suggestion
When rendering imported MDX content, custom components can also be passed via the `components` prop:
```

Nit on colons and `code blocks`

<!-- -->
```

### Custom components

Under the hood, MDX will convert Markdown into HTML components. For example, this blockquote:

```md
> A blockquote with *some* emphasis.
```

will be converted into this HTML:

```html
<blockquote>
  <p>A blockquote with <em>some</em> emphasis.</p>
</blockquote>
```

> But what if you want to specify your own markup for these blockquotes? In the above example, you could create a custom `<Blockquote />` component (in any language) that has either a `<slot />`  component or accepts a `children` prop. MDX provides a way to tap into this rendering and use your own custom components. In the above example, you could create a custom Blockquote component (in any language) that has either a `<slot />`  component or accepts a `children` prop. 

Reviewer 2: ```suggestion
But what if you want to specify your own markup for `<blockquote>` elements? In the above example, you could create a custom `<Blockquote />` component (in any language) that either has a `<slot />` component or accepts a `children` prop.
```
Thanks for incorporating the comments @kylebutts! It looks like there's some copy here twice? So I'll clean this up and then happy for you and @bholmesdev to edit for nuance/reasonableness.

Note, making some editorial decisions based strictly on formatting here, so not meant to disapprove of any previous language. I just don't like how it looks to write plural tags and stuff.
